File: Data Structures & Algorithms/quickSort/submission-0.go
// Definition for a pair.
// type Pair struct {
//     Key   int
//     Value string
// }

/*
type Solution struct {

}

func NewSolution() *Solution {

}
*/

func QuickSort(pairs []Pair) []Pair {
	k := len(pairs)
	if k <= 1 {
		return pairs
	}
	pivot := pairs[k-1]
	e := 0
	for i:= 0; i < k-1; i++ {
		if pairs[i].Key < pivot.Key {
			tmp := pairs[e]
			pairs[e] = pairs[i]
			pairs[i] = tmp
			e++
		}
	}
	pairs[k-1] = pairs[e]
	pairs[e] = pivot
	QuickSort(pairs[:e])
	QuickSort(pairs[e+1:])

	return pairs
}

File: Data Structures & Algorithms/counting-bits/submission-0.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        count = []
        for i in range(n+1):
            tmp = 0
            while i > 0:
                if i & 1 == 1:
                    tmp+=1
                i = i >> 1
            count.append(tmp)
        return count